src/renderers/webgl-fallback/WebGLBackend.js

```diff
@@ -1104,11 +1104,16 @@ class WebGLBackend extends Backend {
 
 					const vp = subCamera.viewport;
 
+					const x = vp.x * pixelRatio;
+					const y = vp.y * pixelRatio;
+					const width = vp.width * pixelRatio;
+					const height = vp.height * pixelRatio;
+
 					state.viewport(
-						Math.floor( vp.x * pixelRatio ),
-						Math.floor( ( renderObject.context.height - vp.height - vp.y ) * pixelRatio ),
-						Math.floor( vp.width * pixelRatio ),
-						Math.floor( vp.height * pixelRatio )
+						Math.floor( x ),
+						Math.floor( renderObject.context.height - height - y ),
+						Math.floor( width ),
+						Math.floor( height )
```

Comment:
This fixes the issue on macOS 👍 .

Reply:
I believe the mentioned issue on the Quest (https://github.com/mrdoob/three.js/pull/30337#issuecomment-2596151898) is something different. I'll verify that tomorrow.

Reply:
Maybe it has something to do with the cache buffer I created for the indexes.